### ISSUE TITLE
verify_ssl = False appears to still try cert verification

### ISSUE THREAD

**Issue by dgw** [embolalia/willie#834]
From willie 5.4.0:
````
Connection error: [Errno 1] _ssl.c:504: error:14090086:SSL 
  routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
````
I have verified that my .cfg file still contains `verify_ssl = False`.

Not sure if it's a red herring that willie appears to be trying to use SSLv3 to connect to a server that has only TLSv1+ enabled (as is the default in ZNC 1.6.0).

**Comment by dgw** [comment]
Same issue occurs using 6.0.0a1:
````
Connecting to localhost:9472...
Connection error: [Errno 1] _ssl.c:504: error:14090086:SSL
  routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed
````

Tweaked my local copy for debugging in `willie/irc.py` to read:
````
    def handle_connect(self):
        if self.config.core.use_ssl and has_ssl:
            print( 'verify_ssl = ' + str(self.config.core.verify_ssl) )
````
and the bot does print `verify_ssl = False` on startup before failing to connect.

**Comment by dgw** [comment]
Applied fix manually to downloaded/extracted 6.0.0a1 tarball and the bot connects.

It appears the bot isn't actually connecting, though, due to the following traceback:
````
Connected.
Traceback (most recent call last):
  File "/home/dgw/willie-6.0.0a1/willie/bot.py", line 236, in call
    exit_code = func(willie, trigger)
  File "/home/dgw/willie-6.0.0a1/willie/coretasks.py", line 330, in recieve_cap_list
    recieve_cap_ls_reply(bot, trigger)
  File "/home/dgw/willie-6.0.0a1/willie/coretasks.py", line 384, in recieve_cap_ls_reply
    if bot.config.core.auth_method == 'sasl' or bot.config.core.sasl_password:
AttributeError: 'CoreSection' object has no attribute 'sasl_password'
````
ZNC never showed willie's user as connected, and after a moment this happened:
````
ERROR:willie.modules.willie.irc:ERROR recieved from server: Closing link [Timeout]
Calling shutdown for 0 modules.
Closed!
Warning: Disconnected. Reconnecting in 20 seconds...
````
followed by a couple module class-redefinition warnings and the same `sasl_password` traceback.

**Comment by dgw** [comment]
From IRC:
>&lt;embolalia&gt; it's fixed in master, but I haven't gotten around to another alpha release yet

Closing.
